### PR TITLE
KMS Support

### PR DESCRIPTION
**Pull request by ChaosInTheCRD**
## What this PR does / why we need it
Adds KMS Support, so you can sign and verify with KMS

Description
Before merging docs need to be written. This is also a draft PR as we are waiting for merge on https://github.com/in-toto/go-witness/pull/120.

### CODE REVIEW

**Comment by ChaosInTheCRD** [comment]
TODO:

- [ ] Docs
- [ ] Ensure that appropriate options are available

File: go.mod [hunk 1]
```diff
@@ -1,9 +1,19 @@
 module github.com/in-toto/witness
 
-go 1.19
+go 1.21
+
+toolchain go1.21.4
+
+replace github.com/in-toto/go-witness => ../go-witness
```
Comment (mikhailswift) [inline go.mod:7]: you should checkout go workspaces! https://go.dev/doc/tutorial/workspaces

Comment (adityasaky) [reply go.mod:7]: TIL, this is very handy!

Comment (ChaosInTheCRD) [reply go.mod:7]: 100% - need to put it on my TODOs

**Comment by ChaosInTheCRD** [comment]
Note that tests won't pass until we merge https://github.com/in-toto/go-witness/pull/120 and cut a new release of `go-witness`

**Review by tannerjones4075 — APPROVED** [review]
I reviewed the documentation for KMS and it is thorough and well laid out. I put my stamp of approval :)

**Review by jkjell — COMMENTED** [review]
This is looking really good. Once we cut a `go-witness` release cut and the lint/testing stuff fixed up. I think this will be good to go. I'll dive into `kms.md` more tomorrow.

**Comment by ChaosInTheCRD** [comment]
All tests passing. Hopefully we can get this merged and a new release cut today @jkjell

**Review by jkjell — APPROVED** [review]

**Pull request merged**